packages/astro/src/runtime/client/dev-overlay/overlay.ts

```diff
@@ -1,79 +1,47 @@
 import { getIconElement, isDefinedIcon, type Icon } from './ui-library/icons.js';
-import { DevOverlayTooltip } from './ui-library/tooltip.js';
-import { DevOverlayWindow } from './ui-library/window.js';
 
-type DevOverlayPlugin = DevOverlayPluginDefinition & {
+export type DevOverlayPlugin = DevOverlayPluginDefinition & {
+	builtIn: boolean;
 	active: boolean;
 	status: 'ready' | 'loading' | 'error';
 	eventTarget: EventTarget;
 };
 
-document.addEventListener('DOMContentLoaded', async () => {
-	const WS_EVENT_NAME = 'astro-dev-overlay';
-	const HOVER_DELAY = 750;
-
-	const builtinPlugins: DevOverlayPlugin[] = [
-		astroDevToolPlugin,
-		astroXrayPlugin,
-		astroAuditPlugin,
-	].map((plugin) => ({
-		...plugin,
-		active: false,
-		status: 'loading',
-		eventTarget: new EventTarget(),
-	}));
-
-	const customPluginsImports = (await loadDevOverlayPlugins()) as DevOverlayPluginDefinition[];
-	const customPlugins: DevOverlayPlugin[] = [];
-	customPlugins.push(
-		...customPluginsImports.map((plugin) => ({
-			...plugin,
-			active: false,
-			status: 'loading' as const,
-			eventTarget: new EventTarget(),
-		}))
-	);
-
-	const plugins: DevOverlayPlugin[] = [...builtinPlugins, ...customPlugins];
-
-	for (const plugin of plugins) {
-		plugin.eventTarget.addEventListener('plugin-notification', (evt) => {
-			const target = overlay.shadowRoot?.querySelector(`[data-plugin-id="${plugin.id}"]`);
-			if (!target) return;
-
-			let newState = true;
-			if (evt instanceof CustomEvent) {
-				newState = evt.detail.state ?? true;
-			}
+const WS_EVENT_NAME = 'astro-dev-overlay';
 
-			target.querySelector('.notification')?.toggleAttribute('data-active', newState);
-		});
-	}
+export class AstroDevOverlay extends HTMLElement {
+	shadowRoot: ShadowRoot;
+	hoverTimeout: number | undefined;
+	isHidden: () => boolean = () => this.devOverlay?.hasAttribute('data-hidden') ?? true;
+	devOverlay: HTMLDivElement | undefined;
+	plugins: DevOverlayPlugin[] = [];
+	HOVER_DELAY = 750;
+	hasBeenInitialized = false;
 
-	class AstroDevOverlay extends HTMLElement {
-		shadowRoot: ShadowRoot;
-		hoverTimeout: number | undefined;
-		isHidden: () => boolean = () => this.devOverlay?.hasAttribute('data-hidden') ?? true;
-		devOverlay: HTMLDivElement | undefined;
-
-		constructor() {
-			super();
-			this.shadowRoot = this.attachShadow({ mode: 'open' });
-		}
+	constructor() {
+		super();
+		this.shadowRoot = this.attachShadow({ mode: 'open' });
+	}
 
-		// connect component
-		async connectedCallback() {
+	// Happens whenever the component is connected to the DOM
+	// When view transitions are enabled, this happens every time the view changes
+	async connectedCallback() {
+		if (!this.hasBeenInitialized) {
 			this.shadowRoot.innerHTML = `
     <style>
+			:host {
+				z-index: 999999;
+				view-transition-name: astro-dev-overlay;
+				display: contents;
+			}
+
+			::view-transition-old(astro-dev-overlay),
+			::view-transition-new(astro-dev-overlay) {
+  			animation: none;
+			}
```

Comment:
All of this is required to ensure animations don't affect the dev overlay, otherwise it might move, get faded away etc